tgui/packages/tgui/interfaces/modules/TGUILatheControl.tsx

```diff
@@ -422,14 +448,32 @@ const LatheDesign = (props: LatheDesignProps, context) => {
   const { data, act, moduleID } = useModule<TGUILatheControlData>(context);
 
   // materials: key = material id
+  // mats maps parts to materials. i think? ask kevinz.
   let [mats, setMats] = useLocalState<Record<string, string>>(context, `${moduleID}-${props.design.id}-mats`, {});
   // ingredients: key = ingredient id/value
   let [inds, setInds] = useLocalState<Record<string, string>>(context, `${moduleID}-${props.design.id}-inds`, {});
 
+  if (!areMaterialsChosen(props.design.material_parts || {}, mats) && props.design.autodetect_tags && props.design.material_parts) {
```

Comment:
this should be done in the initializer of useLocalState, not ran every time the ui is rendered

Reply:
this only runs once per design. should be fine then??